src/module/apps/trait-selector.js

```diff
@@ -9,120 +9,164 @@ export class TraitSelectorSFRPG extends FormApplication {
 
         options.id = "trait-selector";
         options.classes = ["sfrpg"];
-        options.title = "Actor Trait Selection";
+        options.title = "Trait Selection";
         options.template = "systems/sfrpg/templates/apps/trait-selector.hbs";
-        options.width = 320;
+        options.width = 480;
         options.height = "auto";
 
         return options;
     }
 
     /**
-     * Return a reference to the target attribute
+     * Provide data to the HTML template for rendering
      *
-     * @type {String}
+     * @returns {Object}
      */
-    get attribute() {
-        return this.options.name;
+    getData() {
+        console.log(this);
+
+        // Initialize variables for easy access
+        const dataFormat = this.options.format;
+        const dataLocation = this.options.location;
+        const traitData = getProperty(this.object, dataLocation);
+
+        // Choose the appropriate data parser to create the form
+        switch (dataFormat) {
```

Comment:
So I think instead of conditionally returning data to the template depending on `format`, each type of format could correspond to a subclass of `TraitSelectorSFRPG`, which declares its own methods for getting and getting its trait choices. For example, the actor trait selector in base.js would instantiate an `ActorTraitSelectorSFRPG`.

Then you could end up with something like this in a "base trait selector`
```js
getData() {
        console.log(this);

        // Initialize variables for easy access
        const dataFormat = this.options.format;
        const dataLocation = this.options.location;
        const traitData = getProperty(this.object, dataLocation);

       return this._getTraitChoices(traitData);
```
The base trait selector doesn't define a `_getTraitChoices`, but each subclass would, so it would call that on the subclass (may require a `super.getData()`).

Reply:
Alright I think I've changed it in the way you suggested. Now there's a subclass for actor traits that extends the base trait selector, and the switch statements are gone. Let me know if that looks better.

Reply:
This does mean that for each sheet type (actor, item) where the trait selector is being called, only that one subclass can be called and there is an assumed data format that can't differ between traits on that sheet.

Unless, of course, you added a switch on the `dataFormat` property within the subclass, which I guess is just how you'd do it if it came to that.

Reply:
Looks good. Yes, the intent is if a single sheet calls multiple types it uses something like this to determine which:
```js
const cls = {
  "actorTraits": ActorTraitSelector,
  "someOtherTrait": SomeOtherTraitSelector
}[options.format];
new cls(options);
```

Reply:
Cool, didn't know you could write it that way. I'll carry on applying it to weapon properties, then.